File: bridge/bridge_notifications.lua
local notifications = {}

function notifications.is_supported()
	return false
end

function notifications.schedule(notification, on_success, on_failure)
	if on_failure then
		on_failure()
	end
end

function notifications.cancel(id, on_success, on_failure)
	if on_failure then
		on_failure()
	end
end

function notifications.cancel_all(on_success, on_failure)
	if on_failure then
		on_failure()
	end
end

return notifications

File: bridge/src/bridge_notifications.cpp
#if defined(DM_PLATFORM_HTML5)
#include "bridge_notifications.h"
#include "bridge.h"

int bridge::notifications::isSupported(lua_State* L) {
    return getBoolean(L, js_bridge_notifications_isSupported);
}

int bridge::notifications::schedule(lua_State* L) {
    return makeCallbackWithJson(L, js_bridge_notifications_schedule, false);
}

int bridge::notifications::cancel(lua_State* L) {
    return makeCallbackWithString(L, js_bridge_notifications_cancel, false);
}

int bridge::notifications::cancelAll(lua_State* L) {
    return makeCallback(L, js_bridge_notifications_cancelAll, false);
}

#endif
